### PR TITLE
State and validation for tag mappings

### PR DESCRIPTION
**Pull request by carvil**
This PR adds `state` and `message` fields into TagMapping objects.

It then validates TagMappings before submitting the links to the Publishing API. The validations are:
- making sure the `content_id` is valid;
- making sure we only submit `taxon` link types for now;
- making sure the taxons already exist in our system.

Finally, it shows the TagMapping states in the import preview screen like this:

screen shot 2016-08-15 at 23 15 54

Trello: https://trello.com/c/wXQYsPTC/98-adding-state-validations-for-individual-tag-mappings

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -6,9 +6,12 @@ en:
...
     missing_output: Is missing the parameter output as TSV
     spreadsheet_download_error: There is a problem downloading the spreadsheet, please make sure the URL works.
-
   tag_import:
     import_created: Spreadsheet uploaded. It will be imported within a few minutes.
     import_refetched: The import will be imported again.
     import_started: The tagging import has started. It can take a few minutes for the tags to be published.
     import_removed: Import has been removed.
+    errors:
+      invalid_content_id: We could not find the associated content id.
```
Comment (tijmenb) [inline config/locales/en.yml]: Perhaps something like "We could not find this URL on GOV.UK" (people won't know content_id).

File: spec/services/tag_importer/links_publisher_spec.rb [hunk 2]
```diff
@@ -0,0 +1,153 @@
+require 'rails_helper'
+
+RSpec.describe TagImporter::LinksPublisher do
```
Comment (tijmenb) [inline spec/services/tag_importer/links_publisher_spec.rb]: I find this test quite hard to follow. Could we make it smaller and do less, or move parts to an integration test? Feels like it's trying to do too much.

File: config/locales/en.yml (hunk 1 above)
Comment (carvil) [inline config/locales/en.yml]: Sounds good, I will change that.

File: spec/services/tag_importer/links_publisher_spec.rb (hunk 2 above)
Comment (carvil) [inline spec/services/tag_importer/links_publisher_spec.rb]: Yup, especially the setup for each scenario...

I will try to make it better without writing an integration test first if that's ok. If that doesn't work I might split the work into more chunks.

**Comment by carvil** [comment]
@tijmenb The last commit 13e75029eadb69104151226fd7bc48122882e8f1 breaks that big class (and test file) into a few pieces so it's easier to test and to see what is going on.

**Comment by tijmenb** [comment]
👍
